### PR TITLE
Story/bods 2431

### PR DESCRIPTION
**Pull request by alvak1**
Changes to load bank holiday information from api into db

Jira:

https://kpmgengineering.atlassian.net/browse/BODS-2431

### CODE REVIEW

File: config/settings/base.py [hunk 1]
```diff
@@ -509,6 +509,10 @@
     "NPTG_IMPORT_URL", default="https://naptan.api.dft.gov.uk/v1/nptg"
 )
 
+BANK_HOLIDAY_API_URL = env(
```
Comment (pritish-ranjan01) [inline config/settings/base.py:512]: Please update .env templates with this entry.

File: transit_odp/timetables/tasks.py [hunk 2]
```diff
@@ -526,3 +531,17 @@ def task_delete_datasets(*args):
...
+def task_load_bank_holidays():
+    """This is a task to load bank holidays from api to BODS database"""
+    logger.info("Starting process to load bank holidays from api")
+    existing_bank_holidays = BankHolidays.objects.all()
+    bank_holidays = get_bank_holidays()
+    logger.info("completed process to load bank holidays from api successfully")
+    bank_holidays_to_insert = list(get_holidays_records_to_insert(bank_holidays))
+    with transaction.atomic():
+        existing_bank_holidays.delete()
+        BankHolidays.objects.bulk_create(bank_holidays_to_insert, 1000)
```
Comment (pritish-ranjan01) [inline transit_odp/timetables/tasks.py:546]: Can we use the existing BATCH_SIZE property

File: transit_odp/timetables/tasks.py [hunk 3]
```diff
@@ -526,3 +531,17 @@ def task_delete_datasets(*args):
...
+@shared_task()
+def task_load_bank_holidays():
+    """This is a task to load bank holidays from api to BODS database"""
+    logger.info("Starting process to load bank holidays from api")
+    existing_bank_holidays = BankHolidays.objects.all()
+    bank_holidays = get_bank_holidays()
+    logger.info("completed process to load bank holidays from api successfully")
+    bank_holidays_to_insert = list(get_holidays_records_to_insert(bank_holidays))
+    with transaction.atomic():
+        existing_bank_holidays.delete()
```
Comment (pritish-ranjan01) [inline transit_odp/timetables/tasks.py:545]: Can we drop all the objects without querying for all objects.

BankHolidays.objects.all().delete()

File: config/settings/base.py (hunk 1 above)
Comment (alvak1) [reply config/settings/base.py:512]: As default value is provided, is the .env entry required?

Comment (pritish-ranjan01) [reply config/settings/base.py:512]: For this to be managed on cloud as env variables. This .env templates serves as an example.

Comment (alvak1) [reply config/settings/base.py:512]: done

File: transit_odp/timetables/tasks.py (hunk 2 above)
Comment (alvak1) [reply transit_odp/timetables/tasks.py]: done

File: transit_odp/timetables/tasks.py (hunk 3 above)
Comment (alvak1) [reply transit_odp/timetables/tasks.py]: done

**Review by pritish-ranjan01 — APPROVED** [review]

**Pull request merged**